python/ray/data/iterator.py

```diff
@@ -56,6 +59,285 @@ def __iter__(self):
+
+        return arrow_table_to_tensors(batch, dtypes=self.dtypes, device=self.device)
+
+
+@DeveloperAPI
+class DefaultNumpyCollateFn(NumpyBatchCollateFn):
+    """Default collate function for converting Numpy batches to PyTorch tensors."""
+
+    def __init__(
+        self,
+        dtypes: Optional[Union["torch.dtype", Dict[str, "torch.dtype"]]] = None,
+        device: Optional[str] = None,
+    ):
+        """Initialize the collate function.
+
+        Args:
+            dtypes: Optional torch dtype(s) for the tensors
+            device: Optional device to place tensors on
+        """
+        super().__init__(dtypes=dtypes, device=device)
+
+    def __call__(self, batch: "pandas.DataFrame") -> "CollatedData":
+        """Convert a Pandas batch to PyTorch tensors.
+
+        Args:
+            batch: Pandas batch to convert
+
+        Returns:
+            Collated PyTorch tensors
+        """
+        from ray.air._internal.torch_utils import (
+            convert_ndarray_batch_to_torch_tensor_batch,
+        )
+
+        return convert_ndarray_batch_to_torch_tensor_batch(
+            batch, dtypes=self.dtypes, device=self.device
+        )
+
+
+@DeveloperAPI
+class DefaultPandasCollateFn(PandasBatchCollateFn):
+    """Default collate function for converting Pandas batches to PyTorch tensors."""
+
+    def __init__(
+        self,
+        dtypes: Optional[Union["torch.dtype", Dict[str, "torch.dtype"]]] = None,
+        device: Optional[str] = None,
+    ):
+        """Initialize the collate function.
+
+        Args:
+            dtypes: Optional torch dtype(s) for the tensors
+            device: Optional device to place tensors on
+        """
+        super().__init__(dtypes=dtypes, device=device)
+
+    def __call__(self, batch: "pandas.DataFrame") -> "CollatedData":
+        """Convert a Pandas batch to PyTorch tensors.
+
+        Args:
+            batch: Pandas batch to convert
+
+        Returns:
+            Collated PyTorch tensors
+        """
+        from ray.air._internal.torch_utils import (
+            convert_pandas_batch_to_torch_tensor_batch,
+        )
+
+        return convert_pandas_batch_to_torch_tensor_batch(
+            batch, dtypes=self.dtypes, device=self.device
+        )
+
+
+@DeveloperAPI
+class DefaultFinalizeFn:
+    """Default finalize function for moving PyTorch tensors to device."""
+
+    def __init__(
+        self,
+        device: Optional[str] = None,
+    ):
+        """Initialize the finalize function.
+
+        Args:
+            device: Optional device to place tensors on
+        """
+        self.device = device
+
+    def __call__(
+        self,
+        batch: Union[
+            "torch.Tensor",
+            List["torch.Tensor"],
+            Dict[str, "torch.Tensor"],
+            Dict[str, List["torch.Tensor"]],
+        ],
+    ) -> Union[
+        "torch.Tensor",
+        List["torch.Tensor"],
+        Dict[str, "torch.Tensor"],
+        Dict[str, List["torch.Tensor"]],
```

Comment:
same as above. only need to handle `Dict[str, List["torch.Tensor"]],` to `Dict[str, "torch.Tensor"],`

Reply:
I think eventually we'd want the default finalize fn to accept a set of "valid input types" that the collate fn returns. The set of valid types could be this union.

Let's just keep the multiple input types for the future? (where collate and finalize are decoupled)

Reply:
good callout. 
Then the input can be a union of valid types or Any, and the output should be `Dict[str, "torch.Tensor"]` or Any